scene/show-labels-on-layer-in-3d/src/main/java/com/esri/samples/show_labels_on_layer_in_3d/ShowLabelsOnLayerIn3dSample.java

```diff
@@ -0,0 +1,138 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package com.esri.samples.show_labels_on_layer_in_3d;
+
+import com.esri.arcgisruntime.arcgisservices.LabelingPlacement;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.labeling.ArcadeLabelExpression;
+import com.esri.arcgisruntime.symbology.ColorUtil;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.layout.StackPane;
+import javafx.scene.paint.Color;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.arcgisservices.LabelDefinition;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.symbology.TextSymbol;
+
+public class ShowLabelsOnLayerIn3dSample extends Application {
+
+  private SceneView sceneView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+
+      // create stack pane and JavaFX app scene
+      StackPane stackPane = new StackPane();
+      Scene fxScene = new Scene(stackPane);
+
+      // set title, size, and add the JavaFX scene to stage
+      stage.setTitle("Show Labels on Layer in 3D");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(fxScene);
+      stage.show();
+
+      // create a scene from an ArcGISOnline portal item
+      var portalItem = new PortalItem(
+        new Portal("https://www.arcgis.com", false), "850dfee7d30f4d9da0ebca34a533c169");
+      ArcGISScene scene = new ArcGISScene(portalItem);
+
+      // set the scene to the sceneview, and add the sceneview to the JavaFX stack pane
+      sceneView = new SceneView();
+      sceneView.setArcGISScene(scene);
+      // create a progress indicator and display it to indicate loading is in progress
+      var progressIndicator = new ProgressIndicator();
+      progressIndicator.setVisible(true);
+      stackPane.getChildren().addAll(sceneView, progressIndicator);
+
+      scene.addDoneLoadingListener(() -> {
+        if (scene.getLoadStatus() == LoadStatus.LOADED) {
+          // remove the progress indicator when the scene loads
+          progressIndicator.setVisible(false);
+
+          // filter through the scene's operational layers to find the layer named "Gas",
+          // and find the "Gas Main" sublayer to apply the label definition to
+          scene.getOperationalLayers().stream()
+            .filter(layer -> layer.getName().equals("Gas"))
+            .flatMap(gasLayer -> gasLayer.getSubLayerContents().stream())
+            .forEach(subLayer -> {
+              if (subLayer.getName().equals("Gas Main")) {
+                FeatureLayer featureLayer = (FeatureLayer) subLayer;
+                featureLayer.setLabelsEnabled(true);
```

Comment:
Is this an essential step in the workflow to showing labels? i.e. is it worth highlighting with a comment?